resources/osl_shell_alias.rb

```diff
@@ -0,0 +1,37 @@
+resource_name :osl_shell_alias
+provides :osl_shell_alias
+unified_mode true
+
+default_action :add
+
+property :command, String, required: [:add]
+property :alias_name, String, name_property: true
+
+action :add do
+  file destination do
+    mode '0755'
+    content <<-EOF.gsub(/^ {6}/, '')
+      #
+      # This file was generated by Chef
+      # Do NOT modify this file by hand!
+      #
+      alias #{new_resource.alias_name}="#{new_resource.command}"
+      EOF
+  end
+end
+
+action :remove do
+  file destination do
+    action :delete
+  end
+end
+
+action_class do
+  def destination
+    "/etc/profile.d/#{filename}.sh"
+  end
+
+  def filename
+    new_resource.alias_name.to_s.gsub(/\s/, '_')
```

Comment:
`.to_s` isnt needed since this property is a String already.

```suggestion
    new_resource.alias_name.gsub(/\s/, '_')
```

Reply:
👍